imap_processing/lo/l1a/lo_l1a.py

```diff
@@ -126,8 +130,63 @@ def add_dataset_attrs(
         Data with attributes added.
     """
     # TODO: may want up split up these if statements into their
-    # own functions
-    if logical_source == "imap_lo_l1a_histogram":
+    #  own functions
+    # Get global attributes
+    dataset.attrs.update(attr_mgr.get_global_attributes(logical_source))
+    # Get attributes for shcoarse and epoch
+    dataset.shcoarse.attrs.update(attr_mgr.get_variable_attributes("shcoarse"))
+    dataset.epoch.attrs.update(attr_mgr.get_variable_attributes("epoch"))
+
+    if logical_source == "imap_lo_l1a_spin":
+        spin = xr.DataArray(
+            data=np.arange(0, 28, dtype=np.uint16),
+            name="spin",
+            dims=["spin"],
+            attrs=attr_mgr.get_variable_attributes("spin"),
+        )
+        spin_label = xr.DataArray(
+            data=spin.values.astype(str),
+            name="spin_label",
+            dims=["spin_label"],
+            attrs=attr_mgr.get_variable_attributes("spin_label"),
+        )
+
+        dataset = dataset.assign_coords(spin=spin, spin_label=spin_label)
+        dataset.num_completed.attrs.update(
+            attr_mgr.get_variable_attributes("num_completed")
+        )
+        dataset.acq_start_sec.attrs.update(
+            attr_mgr.get_variable_attributes("acq_start_sec")
+        )
+        dataset.acq_start_subsec.attrs.update(
+            attr_mgr.get_variable_attributes("acq_start_subsec")
+        )
+        dataset.acq_end_sec.attrs.update(
+            attr_mgr.get_variable_attributes("acq_end_sec")
+        )
+        dataset.acq_end_subsec.attrs.update(
+            attr_mgr.get_variable_attributes("acq_end_subsec")
+        )
+
+        dataset = dataset.drop_vars(
+            [
+                "version",
+                "type",
+                "sec_hdr_flg",
+                "pkt_apid",
+                "seq_flgs",
+                "src_seq_ctr",
+                "pkt_len",
+                "chksum",
+            ]
+        )
+        # An empty DEPEND_0 is being added to support_data
+        # variables that should only have DEPEND_1
+        # Removing Depend_0 here.
+        # Should look for a fix to this issue
```

Comment:
Add a NOTE or TODO for future searching.
```suggestion
        # TODO: Should look for a fix to this issue
```